src/cript/nodes/supporting_nodes/group.py

```diff
@@ -10,6 +16,196 @@ class Group(BaseNode):
 
     @dataclass(frozen=True)
     class JsonAttributes(BaseNode.JsonAttributes):
+        """
+        all Group attributes
+        """
+
+        node: str = "Group"
+        name: str = ""
+        admins: List[Any] = None
+        users: List[Any] = None
+        notes: str = ""
+
+    _json_attrs: JsonAttributes = JsonAttributes()
+
+    def __init__(self, name: str, admins: List[Any], users: List[Any] = None, **kwargs):
+        """
+        constructor for a Group node
+
+        Parameters
+        ----------
+        name: str
+            group name
+        admins: Any
+            administrator of this group
+        users: List[Any]
+            list of users that are in this Group
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(node="Group")
+        self._json_attrs = replace(self._json_attrs, name=name, admins=admins, users=users)
+        self.validate()
+
+    def validate(self) -> None:
+        """
+        validates Group node
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        CRIPTNodeSchemaError
+        """
         pass
 
-    pass
+    # ------------------ Properties ------------------
+
+    # Group name
+    @property
+    def name(self) -> str:
+        """
+        name property getter method
+
+        Returns
+        -------
+        name: str
+            group name
+        """
+        return self._json_attrs.name
+
+    @name.setter
+    def name(self, new_name: str) -> None:
+        """
+        Setter for the group name
+
+        User cannot set the name for any nodes.
+        Attempt to do so will raise an UneditableAttributeError
+
+        Parameters
+        ----------
+        new_name: str
+            new group name
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        UneditableAttributeError
+            If user attempts to set this property
+        """
+        raise UneditableAttributeError
+
+    # admins
+    @property
+    def admins(self) -> List[Any]:
+        """
+        name property getter method
+
+        Returns
+        -------
+        admins: List[Any]
+            an admin or list of admins
+        """
+        return self._json_attrs.admins
+
+    @admins.setter
+    def admins(self, new_admins: List[Any]) -> None:
```

Comment:
Yeah, we should probably remove this setter.

Reply:
I think there were 2 thoughts on this for when a user tries to edit a node like `User` or `Group` and they cannot do so via the Python SDK
1. raise `AttributeError`
2. raise `UneditableAttributeError`

we could just not implement a setter, which would raise a default python `AttributeError`

To me this is less clear because the user could be sitting there thinking "am I setting it wrong? is there some other method that I must use? is the object wrong." Giving them a clear error that says "You cannot edit this field" makes it much clearer in my opinion

However, I can also see the merit of raising an `AttributeError` as well

Reply:
I see where you are coming from.
I think the right way to do this, is to overload the `__setattr__` function of the BaseNode. And put in some extra logic there, to figure out wheter there is a setter or not.

That allows us to also catch tried modification on attributes that are not part of the data model at all.
And we can raise meaningful exceptions with explanation from there.

Maybe that also allows us to get rid of setters at all

Reply:
Something like:

````
class BaseNode:

  def __setattr__(self, name, value):
       try:
           super().__setattr__(self, name, value)
        except AttributeError as exc:
            raise UneditableError from exc
```

https://stackoverflow.com/questions/37665311/how-do-i-properly-call-the-default-setattr-in-python3